pkg/udf/forward/forward.go

```diff
@@ -78,14 +69,20 @@ type InterStepDataForward struct {
 }
 
 // NewInterStepDataForward creates an inter-step forwarder.
-func NewInterStepDataForward(vertexInstance *dfv1.VertexInstance, fromStep isb.BufferReader, toSteps map[string][]isb.BufferWriter, fsd forwarder.ToWhichStepDecider, mapApplier MapAppliers, fetchWatermark fetch.Fetcher, publishWatermark map[string]publish.Publisher, idleManager wmb.IdleManager, opts ...Option) (*InterStepDataForward, error) {
+func NewInterStepDataForward(vertexInstance *dfv1.VertexInstance, fromStep isb.BufferReader, toSteps map[string][]isb.BufferWriter, fsd forwarder.ToWhichStepDecider, fetchWatermark fetch.Fetcher, publishWatermark map[string]publish.Publisher, idleManager wmb.IdleManager, opts ...Option) (*InterStepDataForward, error) {
 
 	options := DefaultOptions()
 	for _, o := range opts {
 		if err := o(options); err != nil {
 			return nil, err
 		}
 	}
+
+	// verify if initialized with a valid map mode
+	if !isValidMapMode(options) {
```

Comment:
Can we move this validation to the options.go itself? When we set MapMode, we directly set the value and then call this function before returning the options. If we found multiple modes being set, return error. This is more modular.

Reply:
The corner case that occurs with moving this to the function like `WithUDFBatchMap` is that if we pass a value which is nil for the applier then this check will give as a invalid case. 
But some other option might be setting a valid applier later on, so there would be a mismatch.

```
Options [WithUDFBatchMap(nil), WithUDFUnaryMap(applier)]

So in such a case while processing the options, WithUDFBatchMap will error out if we add a check because all appliers will be empty. Even though WithUDFUnaryMap will be setting a correct applier in a later option.
```

Hence, wanted to keep the check only after all options have been set.

Reply:
Can we disallow passing a value which is nil for the applier? If an applier is nil, we don't call With***